yarn-project/noir-contracts/src/contracts/non_native_token_contract/src/main.nr

```diff
@@ -47,17 +47,17 @@ contract NonNativeToken {
         //*********************************/
         amount: pub Field, 
         owner: pub Point,
+        owner_address: pub Field,
         // This field should be hidden
         msg_key: pub Field,
         secret: pub Field,
     ) -> pub [Field; dep::aztec3::abi::PUBLIC_INPUTS_LENGTH] {
         let mut context = PrivateFunctionContext::new();
-        context.args = context.args.push_array([amount, owner.x, owner.y, msg_key, secret]);
-
+        context.args = context.args.push_array([amount, owner.x, owner.y, owner_address, msg_key, secret]);
 
         let mut hash_bytes: [u8; 68] = [0; 68];
         let amount_bytes = amount.to_be_bytes(32);
-        let recipient_bytes = owner.x.to_be_bytes(32);
+        let recipient_bytes = owner_address.to_be_bytes(32);
```

Comment:
@Maddiaa0 mind taking a look at this part?

Reply:
Looks solid to me!